library/control/src/modules/ProductControl.rb

```diff
@@ -1336,16 +1337,47 @@ def RunFrom(from, allow_back)
           end
         end
 
-        Hooks.run("before_#{step_name}")
+        client_name = getClientName(step_name, step_execute)
 
-        result = WFM.CallFunction(getClientName(step_name, step_execute), args)
+        # Check if client exist before continuing
+        if WFM.ClientExists(client_name)
+          Hooks.run("before_#{step_name}")
 
-        # this code will be triggered before the red pop window appears on the user's screen
-        Hooks.run("installation_failure") if result == false
+          result = WFM.CallFunction(client_name, args)
 
-        result = Convert.to_symbol(result)
+          # This code will be triggered before the red pop window appears on the user's screen
+          Hooks.run("installation_failure") if result == false
 
-        Hooks.run("after_#{step_name}")
+          result = result.to_sym
+
+          Hooks.run("after_#{step_name}")
+        else
+          # Client not found. Ask the user if want to continue (related to bsc#1180954)
+          Builtins.y2error("Client '%1' not found", client_name)
```

Comment:
NP: This might be a good time to start using `log.error` (and `include YaST::Logger`, of course) instead of adding more old-style `Builtins.*`. But this is just a suggestion; feel free to ignore.

Reply:
You're completely right, I just wanted to no touch too much that code. But, as well as I did with the popup, using Yast2::Popup, I can start using here the YaST::Logger, as you suggest.